app/models/manageiq/providers/amazon/storage_manager/ebs/cloud_volume_snapshot.rb

```diff
@@ -81,7 +81,17 @@ def delete_snapshot_queue(userid = "system", _options = {})
   def delete_snapshot(_options = {})
     with_provider_object do |snapshot|
       if snapshot
-        snapshot.delete
+        begin
+          snapshot.delete
+        rescue => e
+          notification_options = {
+            :snapshot_op => 'delete',
+            :subject     => "[#{name}]",
+            :error       => e.to_s
+          }
+
+          Notification.create(:type => :vm_snapshot_failure, :options => notification_options)
```

Comment:
yeah, we either need to raise exception or log an error. @gberginc is right, raising MiqVolumeSnapshotDeleteError will be the best.